EasyPost.Tests/InsuranceTest.cs

```diff
@@ -0,0 +1,145 @@
+        private Dictionary<string, object> _parcelData { get; set; }
+        private Dictionary<string, object> _toAddressData { get; set; }
+        private Dictionary<string, object> InsuranceData { get; set; }
+
+        [TestInitialize]
+        public void Initialize()
+        {
+            ClientManager.SetCurrent("NvBX2hFF44SVvTPtYjF0zQ");
+            _fromAddressData = new Dictionary<string, object>
+            {
+                {
+                    "company", "Simpler Postage Inc"
+                },
+                {
+                    "street1", "164 Townsend St"
+                },
+                {
+                    "street2", "Unit 1"
+                },
+                {
+                    "city", "San Francisco"
+                },
+                {
+                    "state", "CA"
+                },
+                {
+                    "zip", "94107"
+                },
+                {
+                    "country", "US"
+                }
+            };
+            _toAddressData = new Dictionary<string, object>()
+            {
+                {
+                    "company", "The White House"
+                },
+                {
+                    "street1", "1600 Pennsylvania Avenue NW"
+                },
+                {
+                    "street2", "DC 20500"
+                },
+                {
+                    "city", "Washington"
+                },
+                {
+                    "state", "DC"
+                },
+                {
+                    "zip", "20500"
+                },
+                {
+                    "country", "US"
+                }
+            };
+            _parcelData = new Dictionary<string, object>
+            {
+                {
+                    "weight", 8.0
+                }
+            };
+
+            Shipment shipment = Shipment.Create(new Dictionary<string, object>
+            {
+                {
+                    "from_address", _fromAddressData
+                },
+                {
+                    "to_address", _toAddressData
+                },
+                {
+                    "parcel", _parcelData
+                },
+                {
+                    "service", "First"
+                },
+                {
+                    "carrier_accounts", new List<string>
+                    {
+                        {
+                            "ca_7642d249fdcf47bcb5da9ea34c96dfcf"
+                        }
+                    }
+                }
+            });
+
+            InsuranceData = new Dictionary<string, object>
+            {
+                {
+                    "to_address", _toAddressData
+                },
+                {
+                    "from_address", _fromAddressData
+                },
+                {
+                    "tracking_code", shipment.tracking_code
+                },
+                {
+                    "amount", (float)100
```

Comment:
Why are we converting **int** to **float** here? Can we just define it as float?

Reply:
Without the cast, it thinks it's an `int`. With decimals, it thinks it's a `double`

Reply:
Resolved by using `decimal` type in [76da6a1](https://github.com/EasyPost/easypost-csharp/pull/205/commits/76da6a1c7800ac4df98f8c230458ee78540999bc)